readthedocs_build/config/config.py

```diff
@@ -397,9 +396,9 @@ def validate_conf_file(self):
         return True
 
     def validate_formats(self):
-        if 'formats' not in self.raw_config or self.raw_config['formats'] is None:
+        _formats = self.raw_config.get('formats')
+        if 'formats' not in self.raw_config or _formats == []:
```

Comment:
I think you can re-write this like: `if not _formats:`. The possible values are

* None, evals to False
* [], evals to False

any different value than that, won't enter in the if clause.

Reply:
I didn't understand how your reply is related to my suggestion 😅

Reply:
With `formats: null` there will be no errors (None), and should be an error since formats always is a list (an empty list, it's no present or is a list with valid options), but never just a single value.

Reply:
Still not following you. I understand how this need to work but, doesn't my suggestion handle the same logic that the code you wrote and is easier to read?

Reply:
Not, because with the get call, if the key is not found it still return None and when there is not present we should let the config pass (this means default values on the rtd code). But if the user has an explicit `None`, that's an error.

Reply:
Here are two tests

https://github.com/rtfd/readthedocs-build/blob/5becfec9784bc422e3a517e93c36ceb4e6ea5a80/readthedocs_build/config/test_config.py#L355-L360

https://github.com/rtfd/readthedocs-build/blob/5becfec9784bc422e3a517e93c36ceb4e6ea5a80/readthedocs_build/config/test_config.py#L321-L324